pkg/client/typed_client.go

```diff
@@ -123,15 +125,40 @@ func (c *typedClient) Patch(ctx context.Context, obj Object, patch Patch, opts .
 	patchOpts.ApplyOptions(opts)
 
 	return o.Patch(patch.Type()).
-		NamespaceIfScoped(o.GetNamespace(), o.isNamespaced()).
+		NamespaceIfScoped(o.namespace, o.isNamespaced()).
 		Resource(o.resource()).
-		Name(o.GetName()).
+		Name(o.name).
 		VersionedParams(patchOpts.AsPatchOptions(), c.paramCodec).
 		Body(data).
 		Do(ctx).
 		Into(obj)
 }
 
+func (c *typedClient) Apply(ctx context.Context, obj runtime.ApplyConfiguration, opts ...ApplyOption) error {
+	o, err := c.resources.getObjMeta(obj)
```

Comment:
At which point does this fail if someone forgets to set the FieldManager? Should we have local validation?

Reply:
It will fail at the kas. The drawback of validating this locally is that we would need to make sure we return the exact same error as the KAS. Given  that this is an error path so hopefully never hot and that we don't validate anything else in the client before sending it off even if we could (e.G that `name` is not-empty when trying to fetch an object) I am not in favor of validating it locally.

Reply:
Makes sense